# --- SDE-COPYRIGHT-NOTE-BEGIN ---
# This copyright note is auto-generated by ./scripts/Create-CopyPatch.
# 
# Filename: package/.../kdebindings/kdebindings-3.5.6-build-against-python-before-2.5.patch
# Copyright (C) 2007 The OpenSDE Project
# 
# More information can be found in the files COPYING and README.
# 
# This patch file is dual-licensed. It is available under the license the
# patched project is licensed under, as long as it is an OpenSource license
# as defined at http://www.opensource.org/ (e.g. BSD, X11) or under the terms
# of the GNU General Public License as published by the Free Software
# Foundation; either version 2 of the License, or (at your option) any later
# version.
# --- SDE-COPYRIGHT-NOTE-END ---

Index: python/pykde/sip/kio/global.sip
===================================================================
--- ./python/pykde/sip/kio/global.sip	(Revision 626831)
+++ ./python/pykde/sip/kio/global.sip	(Revision 626832)
@@ -675,7 +675,7 @@
     QMap<QString,QString> *cppmap = new QMap<QString,QString>;
 
     PyObject *aelem, *belem;
-    ssize_t pos = 0;
+    _SIP_SSIZE_T pos = 0;
     QString *acpp;
     QString *bcpp;
 
Index: python/pykde/sip/kio/authinfo.sip
===================================================================
--- ./python/pykde/sip/kio/authinfo.sip	(Revision 626831)
+++ ./python/pykde/sip/kio/authinfo.sip	(Revision 626832)
@@ -176,7 +176,7 @@
     QMap<QString, QStringList> *map = new QMap<QString, QStringList>;
 
     PyObject *key, *value;
-    ssize_t pos = 0;
+    _SIP_SSIZE_T pos = 0;
     QString *sKey;
     QStringList *sData;
 
Index: python/pykde/sip/kio/kservicetype.sip
===================================================================
--- ./python/pykde/sip/kio/kservicetype.sip	(Revision 626831)
+++ ./python/pykde/sip/kio/kservicetype.sip	(Revision 626832)
@@ -184,7 +184,7 @@
     QMap<QString,QVariant::Type> *cppmap = new QMap<QString,QVariant::Type>;
 
     PyObject *aelem, *belem;
-    ssize_t pos = 0;
+    _SIP_SSIZE_T pos = 0;
     QString *acpp;
     QVariant::Type bcpp;
 
Index: python/pykde/sip/kmdi/kmdichildfrm.sip
===================================================================
--- ./python/pykde/sip/kmdi/kmdichildfrm.sip	(Revision 626831)
+++ ./python/pykde/sip/kmdi/kmdichildfrm.sip	(Revision 626832)
@@ -245,7 +245,7 @@
     QDict<QWidget::FocusPolicy>* qdict = new QDict<QWidget::FocusPolicy>;
 
     PyObject *key, *value;
-    ssize_t pos = 0;
+    _SIP_SSIZE_T pos = 0;
     QString *a0;
     QWidget::FocusPolicy a1;
 
Index: python/pykde/sip/kdeui/kkeydialog.sip
===================================================================
--- ./python/pykde/sip/kdeui/kkeydialog.sip	(Revision 626831)
+++ ./python/pykde/sip/kdeui/kkeydialog.sip	(Revision 626832)
@@ -231,7 +231,7 @@
     QDict<int>* qdict = new QDict<int>;
 
     PyObject *key, *value;
-    ssize_t pos = 0;
+    _SIP_SSIZE_T pos = 0;
     QString *a0;
     int a1;
 
Index: python/pykde/sip/kparts/browserextension.sip
===================================================================
--- ./python/pykde/sip/kparts/browserextension.sip	(Revision 626831)
+++ ./python/pykde/sip/kparts/browserextension.sip	(Revision 626832)
@@ -404,7 +404,7 @@
     QMap<QCString,QCString> *cppmap = new QMap<QCString,QCString>;
 
     PyObject *aelem, *belem;
-    ssize_t pos = 0;
+    _SIP_SSIZE_T pos = 0;
     QCString *acpp;
     QCString *bcpp;
 
@@ -487,7 +487,7 @@
     QMap<QCString,int> *cppmap = new QMap<QCString,int>;
 
     PyObject *aelem, *belem;
-    ssize_t pos = 0;
+    _SIP_SSIZE_T pos = 0;
     QCString *acpp;
     int bcpp;
 
Index: python/pykde/sip/kdecore/bytearray.sip
===================================================================
--- ./python/pykde/sip/kdecore/bytearray.sip	(Revision 626831)
+++ ./python/pykde/sip/kdecore/bytearray.sip	(Revision 626832)
@@ -620,7 +620,7 @@
     QMap<QCString,DCOPRef> *cppmap = new QMap<QCString,DCOPRef>;
 
     PyObject *aelem, *belem;
-    ssize_t pos = 0;
+    _SIP_SSIZE_T pos = 0;
     QCString *acpp;
     DCOPRef *bcpp;
 
@@ -705,7 +705,7 @@
     QMap<QString,DCOPRef> *cppmap = new QMap<QString,DCOPRef>;
 
     PyObject *aelem, *belem;
-    ssize_t pos = 0;
+    _SIP_SSIZE_T pos = 0;
     QString *acpp;
     DCOPRef *bcpp;
 
Index: python/pykde/sip/kdecore/kconfigdata.sip
===================================================================
--- ./python/pykde/sip/kdecore/kconfigdata.sip	(Revision 626831)
+++ ./python/pykde/sip/kdecore/kconfigdata.sip	(Revision 626832)
@@ -141,7 +141,7 @@
     QMap<KEntryKey,KEntry> *cppmap = new QMap<KEntryKey,KEntry>;
 
     PyObject *aelem, *belem;
-    ssize_t pos = 0;
+    _SIP_SSIZE_T pos = 0;
     KEntryKey *acpp;
     KEntry *bcpp;
 
Index: python/pykde/sip/kdecore/kaccel.sip
===================================================================
--- ./python/pykde/sip/kdecore/kaccel.sip	(Revision 626831)
+++ ./python/pykde/sip/kdecore/kaccel.sip	(Revision 626832)
@@ -209,7 +209,7 @@
     QMap<int,QString> *map = new QMap<int,QString>;
 
         PyObject *key, *value;
-        ssize_t pos = 0;
+        _SIP_SSIZE_T pos = 0;
         int iKey;
         QString *sData;
 
Index: python/pykde/sip/kdecore/kcompletion.sip
===================================================================
--- ./python/pykde/sip/kdecore/kcompletion.sip	(Revision 626831)
+++ ./python/pykde/sip/kdecore/kcompletion.sip	(Revision 626832)
@@ -201,7 +201,7 @@
     QMap<KCompletionBase::KeyBindingType,KShortcut> *kbmap = new QMap<KCompletionBase::KeyBindingType,KShortcut>;
 
     PyObject *key, *value;
-        ssize_t pos = 0;
+        _SIP_SSIZE_T pos = 0;
         KShortcut *cValue;
         int iserr;
 
Index: python/pykde/sip/kdecore/kmacroexpander.sip
===================================================================
--- ./python/pykde/sip/kdecore/kmacroexpander.sip	(Revision 626831)
+++ ./python/pykde/sip/kdecore/kmacroexpander.sip	(Revision 626832)
@@ -169,7 +169,7 @@
     QMap<QChar,QString> *map = new QMap<QChar,QString>;
 
     PyObject *key, *value;
-    ssize_t pos = 0;
+    _SIP_SSIZE_T pos = 0;
     QChar *cKey;
     QString *sData;
 
@@ -255,7 +255,7 @@
     QMap<QString, QStringList> *map = new QMap<QString, QStringList>;
 
     PyObject *key, *value;
-    ssize_t pos = 0;
+    _SIP_SSIZE_T pos = 0;
     QString *sKey;
     QStringList *slData;
 
@@ -343,7 +343,7 @@
     QMap<QChar,QStringList> *map = new QMap<QChar,QStringList>;
 
     PyObject *key, *value;
-    ssize_t pos = 0;
+    _SIP_SSIZE_T pos = 0;
     QChar *cKey;
     QStringList *slData;
 
Index: python/pykde/sip/kdecore/kurl.sip
===================================================================
--- ./python/pykde/sip/kdecore/kurl.sip	(Revision 626831)
+++ ./python/pykde/sip/kdecore/kurl.sip	(Revision 626832)
@@ -234,7 +234,7 @@
         void                 __setitem__ (int, const KURL&);
 %MethodCode
 //takes index | (int) | value | (KURL)
-    ssize_t len;
+    _SIP_SSIZE_T len;
 
     len = sipCpp -> count();
 
@@ -248,7 +248,7 @@
 %MethodCode
 //takes range | (a Python slice) | urlList | (KURL.List)
 
-    ssize_t len, start, stop, step, slicelength;
+    _SIP_SSIZE_T len, start, stop, step, slicelength;
 
     len = sipCpp -> count();
 
@@ -278,7 +278,7 @@
         void                 __delitem__ (int);
 %MethodCode
 //takes index | (int)
-    ssize_t len;
+    _SIP_SSIZE_T len;
 
     len = sipCpp -> count();
 
@@ -291,7 +291,7 @@
         void                 __delitem__ (SIP_PYSLICE);
 %MethodCode
 //takes range | (a Python slice)
-    ssize_t len, start, stop, step, slicelength;
+    _SIP_SSIZE_T len, start, stop, step, slicelength;
 
     len = sipCpp -> count();
     if (sipConvertFromSliceObject(a0,len,&start,&stop,&step,&slicelength) < 0)
@@ -308,7 +308,7 @@
 %MethodCode
 //returns (KURL)
 //takes index | (int)
-    ssize_t len;
+    _SIP_SSIZE_T len;
 
     len = sipCpp -> count();
 
@@ -322,7 +322,7 @@
 %MethodCode
 //returns (KURL.List)
 //takes range | (a Python slice)
-    ssize_t len, start, stop, step, slicelength;
+    _SIP_SSIZE_T len, start, stop, step, slicelength;
 
     len = sipCpp -> count();
 
Index: python/pykde/sip/kdecore/kconfig.sip
===================================================================
--- ./python/pykde/sip/kdecore/kconfig.sip	(Revision 626831)
+++ ./python/pykde/sip/kdecore/kconfig.sip	(Revision 626832)
@@ -158,7 +158,7 @@
     QMap<QString,QString> *cppmap = new QMap<QString,QString>;
 
     PyObject *aelem, *belem;
-    ssize_t  pos = 0;
+    _SIP_SSIZE_T  pos = 0;
     QString *acpp;
     QString *bcpp;
 
Index: python/pyqt/sip/qt/qstring.sip
===================================================================
--- ./python/pyqt/sip/qt/qstring.sip	(Revision 626831)
+++ ./python/pyqt/sip/qt/qstring.sip	(Revision 626832)
@@ -746,7 +746,7 @@
 
 	QString operator[](SIP_PYSLICE) const;
 %MethodCode
-		ssize_t len, start, stop, step, slicelength;
+		_SIP_SSIZE_T len, start, stop, step, slicelength;
 
 		len = sipCpp -> length();
 
Index: python/pyqt/sip/qt/qdir.sip
===================================================================
--- ./python/pyqt/sip/qt/qdir.sip	(Revision 626831)
+++ ./python/pyqt/sip/qt/qdir.sip	(Revision 626832)
@@ -238,7 +238,7 @@
 
 	QString operator[](int) const;
 %MethodCode
-		ssize_t len;
+		_SIP_SSIZE_T len;
 
 		Py_BEGIN_ALLOW_THREADS
 		len = sipCpp -> count();
@@ -256,7 +256,7 @@
 
 	QStringList operator[](SIP_PYSLICE) const;
 %MethodCode
-		ssize_t len, start, stop, step, slicelength;
+		_SIP_SSIZE_T len, start, stop, step, slicelength;
 
 		Py_BEGIN_ALLOW_THREADS
 		len = sipCpp -> count();
Index: python/pyqt/sip/qt/qstringlist.sip
===================================================================
--- ./python/pyqt/sip/qt/qstringlist.sip	(Revision 626831)
+++ ./python/pyqt/sip/qt/qstringlist.sip	(Revision 626832)
@@ -99,7 +99,7 @@
 
 	void __setitem__(int,const QString &);
 %MethodCode
-		ssize_t len;
+		_SIP_SSIZE_T len;
 
 		len = sipCpp -> count();
 
@@ -111,7 +111,7 @@
 
 	void __setitem__(SIP_PYSLICE,const QStringList &);
 %MethodCode
-		ssize_t len, start, stop, step, slicelength;
+		_SIP_SSIZE_T len, start, stop, step, slicelength;
 
 		len = sipCpp -> count();
 
@@ -142,7 +142,7 @@
 
 	void __delitem__(int);
 %MethodCode
-		ssize_t len;
+		_SIP_SSIZE_T len;
 
 		len = sipCpp -> count();
 
@@ -154,7 +154,7 @@
 
 	void __delitem__(SIP_PYSLICE);
 %MethodCode
-		ssize_t len, start, stop, step, slicelength;
+		_SIP_SSIZE_T len, start, stop, step, slicelength;
 
 		len = sipCpp -> count();
 
@@ -170,7 +170,7 @@
 
 	QString operator[](int);
 %MethodCode
-		ssize_t len;
+		_SIP_SSIZE_T len;
 
 		len = sipCpp -> count();
 
@@ -182,7 +182,7 @@
 
 	QStringList operator[](SIP_PYSLICE);
 %MethodCode
-		ssize_t len, start, stop, step, slicelength;
+		_SIP_SSIZE_T len, start, stop, step, slicelength;
 
 		len = sipCpp -> count();
 
Index: dcoppython/shell/marshaller.cpp
===================================================================
--- ./dcoppython/shell/marshaller.cpp	(Revision 626831)
+++ ./dcoppython/shell/marshaller.cpp	(Revision 626832)
@@ -24,6 +24,10 @@
 
 #include <kurl.h>
 
+#if PY_VERSION_HEX < 0x02050000
+typedef int Py_ssize_t;
+#endif
+
 namespace PythonDCOP {
 
 #include "marshal_funcs.h"
@@ -124,7 +128,7 @@
     if (!PyDict_Check(obj)) return false;
 
 
-    ssize_t c=0;
+    Py_ssize_t c=0;
     PyObject *key, *val;
     while (PyDict_Next(obj, &c, &key, &val)==1)
       if (!key_type.isMarshallable(key) ||
